### PR TITLE
envoy 1.29.0 (with patch for cel-cpp linker issue)

### PR DESCRIPTION
**Pull request by timsutton**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

It isn't clear whether the fix in https://github.com/envoyproxy/envoy/pull/32148 would land in its current form, but testing it out with a copied patch.

### CODE REVIEW

**Comment by timsutton** [comment]
This would need the `long-build` label I think

**Comment by timsutton** [comment]
@p-linnane @chenrui333 if you have a moment to help me with adding the above label for this one?

File: Formula/e/envoy.rb [hunk 1]
```diff
@@ -43,6 +43,10 @@ class Envoy < Formula
     cause "C++17 support and tcmalloc requirement"
   end
 
+  # Remove this patch when this is landed (or resolved) in a release:
+  # https://github.com/envoyproxy/envoy/pull/32148
```
Comment (timsutton) [inline Formula/e/envoy.rb:47]: ^ now merged into envoyproxy/envoy main

Comment (ggreenway) [reply Formula/e/envoy.rb:47]: It's been backported onto 1.29 (https://github.com/envoyproxy/envoy/commit/2cc41dcd321dab2356f29cd5902db65c97675291), so you should probably use that commit so the patch doesn't have to be copied to this repo

Comment (timsutton) [reply Formula/e/envoy.rb:47]: Ah! I didn't realize. Thank you, will update it!

**Comment by p-linnane** [comment]
@timsutton Is this ready for a rerun?

**Comment by timsutton** [comment]
@p-linnane Yes please!

**Comment by timsutton** [comment]
@p-linnane wait, hold up a minute 🤦‍♂️ – need to still delete the old hardcoded patch

**Comment by timsutton** [comment]
..and I see now that it's now outdated anyway, 1.29.1 has CVE fixes and includes the patch. Let me just open up a separate PR for 1.29.1, we can close this PR if that one's green.

**Comment by timsutton** [comment]
Going to just pre-emptively close in favor of https://github.com/Homebrew/homebrew-core/pull/162277

**Pull request closed without merging**